lib/games_collection.rb

```diff
@@ -92,4 +101,102 @@ def average_goals_by_season
       (total_goals.to_f / games_count).round(2)
     end
   end
+
+  def games_by_team(team_id)
+    win = Hash.new {|h, k| h[k] = {wins: 0, total: 0}}
+    @games.each do |game|
+      if game.away_team_id == team_id && game.winner == "away"
+        win[game.season][:wins] += 1
+        win[game.season][:total] += 1
+      elsif game.home_team_id == team_id && game.winner == "home"
+        win[game.season][:wins] += 1
+        win[game.season][:total] += 1
+      elsif game.away_team_id == team_id || game.home_team_id == team_id
+        win[game.season][:total] += 1
+      end
+    end
+    win
+  end
+
+  def best_season(team_id)
+    games_by_team(team_id).max_by do |season, numbers|
+      numbers[:wins].to_f / numbers[:total]
+    end.first
+  end
+
+  def worst_season(team_id)
+    games_by_team(team_id).min_by do |season, numbers|
+      numbers[:wins].to_f / numbers[:total]
+    end.first
+  end
+
+  def average_win_percentage(team_id)
```

Comment:
Since we are using hashes to count other values it may be worth refactoring this to follow a similar pattern instead of having two separate counters